blockchain/message_pool/src/msgpool.rs

```diff
@@ -467,13 +445,14 @@ where
     /// Load local messages into pending. As of  right now messages are not deleted from self's
     /// local_message field, possibly implement this in the future?
     pub async fn load_local(&mut self) -> Result<(), Error> {
-        for (key, value) in self.local_msgs.clone() {
-            self.add(&value).await.unwrap_or_else(|err| {
+        let msg_vec = self.local_msgs.clone();
```

Comment:
shouldn't have to clone

Reply:
clone has to be there, if it isn't the borrow checker complains when calling self.add()

Reply:
there are ways to get around this, such as replacing the loop with something like:
```
        while let Some(msg) = self.local_msgs.pop() {
```

But I noticed also they are persisting these messages, and wonder why they are all kept in memory anyway, any specific reason?

Reply:
I don't know either, it would make more sense to just remove the messages from pending when they get loaded.